aten/src/ATen/native/sparse/SparseTensorMath.cpp

```diff
@@ -153,14 +154,26 @@ SparseTensor pow_sparse_scalar(const SparseTensor& t, Scalar value) {
 SparseTensor& div_out_sparse_zerodim(SparseTensor& r, const SparseTensor& t, const Tensor& value);
 
 Tensor div_sparse(const Tensor& self, const Tensor& value) {
-  Tensor result = at::empty({0}, self.options());
+  auto commonDtype = at::result_type(self, value);
+  Tensor result = at::empty({0}, self.options().dtype(commonDtype));
   return div_out_sparse_zerodim(result, self, value);
 }
 
 Tensor& div_sparse_(Tensor& self, const Tensor& value) {
   return div_out_sparse_zerodim(self, self, value);
 }
 
+// inplace coalesce
+const SparseTensor& coalesce_(const SparseTensor& tensor) {
```

Comment:
(carrying over from previous discussions) We should either:
- document that in-place division does a copy
- file an issue for it.

This is because, by definition, in-place operations save memory by not doing a copy. This is not explicitly stated in our documentation, but I think that is a falling in our documentation.

Reply:
Filed an issue, I think it's out of scope for this PR since the documentation issue is also related to type promoting ops more general, not just for sparse.  E.g. if you do a float_tensor *= double_tensor it's going to create a temporary double result tensor, which it will then copy into the float tensor.

Ticket here:
https://github.com/pytorch/pytorch/issues/30533